.github/workflows/windows_intel_oneAPI.yml

```diff
@@ -47,17 +47,23 @@ jobs:
         with:
           submodules: recursive
 
-      - name: Setup conda
-        uses: s-weigand/setup-conda@678f22c807cb6fde6a290be6f3546877c98ec66f # v1.2.2
+      - name: Setup Conda
+        uses: conda-incubator/setup-miniconda@a4260408e20b96e80095f42ff7f1a15b27dd94ca # v3.0.4
         with:
-          update-conda: true
           python-version: 3.11
-      - run: conda --version
+          channels: conda-forge
+          channel-priority: true
+          use-only-tar-bz2: false
+          miniforge-variant: Miniforge3
+          miniforge-version: latest
+          use-mamba: true
+
+      - run: mamba --version
       - run: which python
 
       - name: Install packages from conda
         run: |
-          conda install -c conda-forge pkg-config meson meson-python ninja openblas libblas=*=*openblas numpy==2.0 cython pybind11 pytest pytest-xdist pytest-timeout pytest-fail-slow pooch rich-click click doit pydevtool hypothesis
+          mamba install -c conda-forge pkg-config meson meson-python ninja openblas libblas=*=*openblas numpy==2.0 cython pybind11 pytest pytest-xdist pytest-timeout pytest-fail-slow pooch rich-click click doit pydevtool hypothesis
```

Comment:
I don't think we need to use mamba here, miniforge should be using it by default.

Should specification of the channel be necessary if we do the spec in the setup-miniconda action?

In numpy we seem to be creating and then activating an environment on each part of the job, see [here]( https://github.com/numpy/numpy/blob/main/.github/workflows/macos.yml#L94).

Also, I wonder if we need to activate the [base environment](https://github.com/conda-incubator/setup-miniconda?tab=readme-ov-file#activate-base-environment)? This will involve:

```
- uses: conda-incubator/setup-miniconda@v3
  with:
    auto-activate-base: true
    activate-environment: ""
```